fitbenchmarking/parsing/cutest_parser.py

```diff
@@ -100,19 +104,48 @@ def _function(self, x, *params):
         """
         os.environ["MASTSIF"] = self.mastsif_dir.name
 
-        for cx, cf in self._cache:
+        for cx, cf in self._cache_f:
             if np.isclose(cx, x).all():
                 f = cf
                 break
         else:
             fname, _, _, _ = self._setup_data(x)
             p = _import_problem(fname)
             f = p.objcons
-            self._cache.append((x, f))
+            self._cache_f.append((x, f))
         _, fx = f(np.asarray(params))
 
         return fx
 
+    def _jacobian(self, x, *params):
+        """
+        If these x values have been passed in before, then run the function to
+        calculate the Jacobian. Otherwise, create a new problem file, parse and
+        cache the Jacobian
+
+        :param x: The data to evaluate at
+        :type x: np.array
+        :return: The result of evaluating at the given x
+        :rtype: np.array
+        """
+        os.environ["MASTSIF"] = self.mastsif_dir.name
+
+        for cx, cg in self._cache_g:
+            if np.isclose(cx, x).all():
+                g = cg
+                break
+        else:
+            fname, _, _, _ = self._setup_data(x)
+            p = _import_problem(fname)
+            g = p.lagjac
+            self._cache_g.append((x, g))
+        _, gx = g(np.asarray(params)[0])
+
+        # We negate the gradient since `min ||r(x)|| s.t.
+        # `r(x) = f(x)-y = y-f(x)` it seems that CUTEst uses the
+        # negation of our implementation of `r(x)`
+        return -gx
```

Comment:
This needs to also take into account weights.  If `use_errors = yes`, then we're minimizing f(x) = ||(y - f(x))/e||, and so we need to divide the derivatives of f by the errors.

Reply:
It appears that `use_errors` is not used in any of the function evaluations from what I have looked at